server/src/Dao/RevisionDao.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Dao;
+
+use Doctrine\DBAL\Connection;
+
+class RevisionDao extends AbstractDao {
+
+	/**
+	 * @param string[] $users
+	 * @param null $startDate
+	 * @param null $endDate
+	 * @param int $limit
+	 * @param null $continue
+	 * @return array
+	 */
+	public function getUserRevisionsInCommonPages(
+		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
+	) {
+		// build subquery to find common pages between users
+		$subQuery = $this->conn->createQueryBuilder();
+		$subQuery->select( 'rev_page' )
+			->from( 'revision_userindex' )
+			->where( 'rev_user_text in (:users)' )
+			->groupBy( 'rev_page' )
+			->having( 'count(distinct rev_user) > 1' );
+
+		// build query to find interaction between users in common pages
+		$query = $this->conn->createQueryBuilder();
+		$query->select( 'rev_id' )
+			->from( 'revision_userindex', 'r' )
+			->where( 'rev_user_text in (:users)' )
+			->andWhere( $query->expr()->in( 'rev_page', $subQuery->getSQL() ) )
```

Comment:
Was the subquery faster/better than the JOIN?

Reply:
It was by a lot. It is hard to debug queries to the table views so i can't say why, but it was by a lot. Tested it with users with low edit count and took way longer.

Reply:
fascinating.